zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -347,62 +323,82 @@ benchmarks! {
 
     internal_resolve_categorical_reported {
         let categories : u16 = T::MaxCategories::get().saturated_into();
-        let (_, market_id) =
-            setup_resolve_common_categorical_after_dispute::<T>(0, 0, categories)?;
+        let (_, market_id) = setup_resolve_common_categorical::<T>(0, 0, categories)?;
     }: {
         let market = T::MarketCommons::market(&market_id)?;
-        let disputes = crate::Disputes::<T>::get(market_id);
-        T::SimpleDisputes::on_resolution(&disputes, &market_id, &market)?
+        Pallet::<T>::on_resolution(&market_id, &market)?;
+    } verify {
+        let market = T::MarketCommons::market(&market_id)?;
+        assert_eq!(market.status, MarketStatus::Resolved);
     }
 
     internal_resolve_categorical_disputed {
         // d = num. disputes
         let d in 0..T::MaxDisputes::get();
 
         let categories = T::MaxCategories::get();
-        let (caller, market_id) =
-            setup_resolve_common_categorical_after_dispute::<T>(0, 0, categories)?;
+        let (caller, market_id) = setup_resolve_common_categorical::<T>(0, 0, categories)?;
+        T::MarketCommons::mutate_market(&market_id, |market| {
+            let admin = account("admin", 0, 0);
+            market.dispute_mechanism = MarketDisputeMechanism::Authorized(admin);
+            Ok(())
+        })?;
 
         let categories : u32 = categories.saturated_into();
         for i in 0..categories.min(d) {
             let origin = caller.clone();
-            let disputes = crate::Disputes::<T>::get(market_id);
-            let market = T::MarketCommons::market(&Default::default()).unwrap();
-            T::SimpleDisputes::on_dispute(&disputes, &market_id, &market)?;
+            Pallet::<T>::dispute(
+                RawOrigin::Signed(origin).into(),
+                market_id,
+                OutcomeReport::Categorical(i.saturated_into::<u16>()),
+            )?;
         }
     }: {
         let market = T::MarketCommons::market(&market_id)?;
-        let disputes = crate::Disputes::<T>::get(market_id);
-        T::SimpleDisputes::on_resolution(&disputes, &market_id, &market)?
+        Pallet::<T>::on_resolution(&market_id, &market)?;
+    } verify {
+        let market = T::MarketCommons::market(&market_id)?;
```

Comment:
The market should be queried in the setup {} (first) section one time.

Reply:
in verify section I just wanted to assert that after call to on_resolution() market status is properly updated.

Reply:
OK